stat.go

```diff
@@ -648,6 +648,47 @@ func KullbackLeibler(p, q []float64) float64 {
 	return kl
 }
 
+// OrdinaryLinearRegression computes the simple linear regression of x and y.
+// It returns the values for \hat{\alpha}, \hat{\beta} and the coefficient
+// of determination, r. If origin is true, the regression is forced to pass
+// through the origin.
+func OrdinaryLinearRegression(x, y, weights []float64, origin bool) (alpha, beta, r float64) {
+	w := 1.0
+	yMean := Mean(y, weights)
+	if origin {
+		var x2Sum, xySum float64
+		for i, xi := range x {
+			if weights != nil {
```

Comment:
I'm not convinced that a closure is likely to be better than a condition; the condition will be well predicted while the closure cannot be inlined. I think this is clearer too.